src/app/fddev/configure/genesis.c

```diff
@@ -16,140 +24,136 @@ enabled( config_t * const config ) {
-     user namespace once multi-threaded). We also want all those threads
-     gone once genesis creation completes, but Labs does not clean them up. */
-  pid_t pid = fork();
-  if( FD_UNLIKELY( pid == -1 ) ) FD_LOG_ERR(( "fork() failed (%i-%s)", errno, fd_io_strerror( errno ) ));
-
-  if( FD_LIKELY( !pid ) ) {
-    /* switch to non-root uid/gid for file creation. permissions checks still done as root. */
-    gid_t gid = getgid();
-    uid_t uid = getuid();
-    if( FD_LIKELY( gid == 0 && setegid( config->gid ) ) )
-      FD_LOG_ERR(( "setegid() failed (%i-%s)", errno, fd_io_strerror( errno ) ));
-    if( FD_LIKELY( uid == 0 && seteuid( config->uid ) ) )
-      FD_LOG_ERR(( "seteuid() failed (%i-%s)", errno, fd_io_strerror( errno ) ));
-
-    umask( S_IRWXO | S_IRWXG );
-    fd_ext_genesis_main( (const char **)argv );
-    exit_group( 0 );
+    fd_pod_insert_ulong( pod, "hashes_per_tick", hashes_per_tick );
+
+  } else if( 1UL==config->development.genesis.hashes_per_tick ) {
+
+    /* do not set hashes_per_tick field */
+
   } else {
-    int wstatus;
-    if( FD_UNLIKELY( waitpid( pid, &wstatus, 0 )==-1 ) ) FD_LOG_ERR(( "waitpid() failed (%i-%s)", errno, fd_io_strerror( errno ) ));
-    if( FD_UNLIKELY( WIFSIGNALED( wstatus ) ) )
-      FD_LOG_ERR(( "genesis creation process terminated by signal %i-%s", WTERMSIG( wstatus ), fd_io_strsignal( WTERMSIG( wstatus ) ) ));
-    if( FD_UNLIKELY( WEXITSTATUS( wstatus ) ) )
-      FD_LOG_ERR(( "genesis creation process exited with status %i", WEXITSTATUS( wstatus ) ));
+
+    /* set hashes_per_tick to the specified value */
+    fd_pod_insert_ulong( pod, "hashes_per_tick", config->development.genesis.hashes_per_tick );
+
   }
+
+  fd_pod_insert_ulong( pod, "ticks_per_slot", config->development.genesis.ticks_per_slot );
+  fd_pod_insert_ulong( pod, "target_tick_µs", config->development.genesis.target_tick_duration_micros );
+
+  fd_pod_insert_ulong( pod, "default_funded.cnt",     config->development.genesis.fund_initial_accounts );
+  fd_pod_insert_ulong( pod, "default_funded.balance", config->development.genesis.fund_initial_amount_lamports );
+
+  /* Serialize blob */
+
+  static uchar scratch_smem[ 16<<20UL ];  /* fits at least 32k accounts */
+         ulong scratch_fmem[ 4 ];
+  fd_scratch_attach( scratch_smem, scratch_fmem,
+                     sizeof(scratch_smem), sizeof(scratch_fmem)/sizeof(ulong) );
+
+  static uchar blob[ 16<<20UL ];
+
+  ulong blob_sz = fd_genesis_create( blob, sizeof(blob), pod );
+  if( FD_UNLIKELY( !blob_sz ) ) FD_LOG_ERR(( "Failed to create genesis blob" ));
+
+  FD_LOG_DEBUG(( "Created genesis blob (sz=%lu)", blob_sz ));
+
+  char genesis_path[ PATH_MAX ];
+  FD_TEST( fd_cstr_printf_check( genesis_path, PATH_MAX, NULL, "%s/genesis.bin", config->ledger.path ) );
+  do {
+    FILE * genesis_file = fopen( genesis_path, "w" );
+    FD_TEST( genesis_file );
+    FD_TEST( 1L == fwrite( blob, blob_sz, 1L, genesis_file ) );
+    fclose( genesis_file );
```

Comment:
FD_TEST (we don't assume what fs this is mounted on, and close could conceivably fail)